Clarify how branches are merged back for contributors to pick a target

martinezjavier opened the pull request:
I usually get confused about which branch should be the target of a pull
request. Add some clarifications about how the branches are merged back,
for contributors to decide the branch they should target a pull-request.

Signed-off-by: Javier Martinez Canillas <javierm@redhat.com>

VladimirSlavik commented on CONTRIBUTING.rst:
```diff
@@ -54,6 +54,14 @@ Depending on where you want to make your contribution please choose your correct
+need to open a separate pull-request for ``master`` in that case.
+
+The ``rhel-x`` branches on the other hand are not merged back into ``master``, and so a separate pull request
+has to be opened targeting either for a ``fXX-devel`` branch or ``master``.
```
I think you need to choose one of "targeting either" and "for", but not both at once. (So it's actually xor...)

martinezjavier replied:
Ok

VladimirSlavik commented on CONTRIBUTING.rst:
```diff
@@ -54,6 +54,14 @@ Depending on where you want to make your contribution please choose your correct
+need to open a separate pull-request for ``master`` in that case.
+
+The ``rhel-x`` branches on the other hand are not merged back into ``master``, and so a separate pull request
+has to be opened targeting either for a ``fXX-devel`` branch or ``master``.
```
I'm not sure if that is within the scope here, but we use the labels "port to X" where X = Fedora or RHEL. Having that label on a PR means this should be in the other branch too, but the PR was not yet created.

Arguably, that is of little use to contributors because they can not add these labels.

martinezjavier replied:
> Arguably, that is of little use to contributors because they can not add these labels.

Yeah, I always thought that this kind of labeling for PR was something handled by the Anaconda core developers as a part of the PR review process.

VladimirSlavik reviewed (approved):
Looks good to me. Thank you!

Somewhat ironically, I guess branching doesn't matter for this very PR because people will usually see the docs from master.

jkonecny12 reviewed (changes requested):
That's great improvement, thanks!

Please just change also the table:

```
| Fedora XX                | fXX-devel    |
```
to this:

```
| Fedora XX and Rawhide                | fXX-devel    |
+---------------------------------------------+-----------------+
| Fedora XX only                              | fXX-release |
```

jkonecny12 commented:
/packit build

jkonecny12 commented:
/test

jkonecny12 commented:
/kickstart-test --testtype smoke

poncovka reviewed (approved):
It looks good to me. Thank you!

martinezjavier commented:
> Please just change also the table:

Ok, will do.

jkonecny12 commented:
@martinezjavier hi, should I re-take this PR and finish that or do you want to do that?

martinezjavier commented:
@jkonecny12 @VladimirSlavik @poncovka sorry for the delay, this fell of my radar.

I've addressed the issues you pointed out and pushed a new version of the commit.

VladimirSlavik reviewed (approved):
Thanks, even better.

jkonecny12 commented:
/kickstart-test --testtype smoke

jkonecny12 reviewed (approved):
Looks great to me now! Thanks!

poncovka reviewed (approved):
It looks great. Thank you!

jkonecny12 commented:
/kickstart-test --testtype smoke
